benchmarks/functional_autograd_benchmark/audio_text_models.py

```diff
@@ -0,0 +1,128 @@
+import torch
+import torch.nn as nn
+
+import torchaudio_models as models
+
+from utils import make_functional, load_weights
+
+def get_wav2letter(device):
+    N = 10
+    input_frames = 700
+    vocab_size = 28
+    model = models.Wav2Letter(num_classes=vocab_size)
+    criterion = torch.nn.NLLLoss()
+    model.to(device)
+    params, names = make_functional(model)
+
+    inputs = torch.rand([N, 1, input_frames], device=device)
+    labels = torch.rand(N, 3, device=device).mul(vocab_size).long()
+
+    def forward(*new_params):
+        load_weights(model, names, new_params)
+        out = model(inputs)
+
+        loss = criterion(out, labels)
+        return loss
+
+    return forward, params
+
+def get_deepspeech(device):
+    supported_rnns = {
+        'lstm': nn.LSTM,
+        'rnn': nn.RNN,
+        'gru': nn.GRU
+    }
+    sample_rate = 16000
+    window_size = 0.02
+    window = "hamming"
+    audio_conf = dict(sample_rate=sample_rate,
+                      window_size=window_size,
+                      window=window,
+                      noise_dir=None)
+
+
+    N = 10
+    num_classes = 10
+    spectrogram_size = 161
+    # Commented are the original sizes in the code
+    seq_length = 500  # 1343
+    target_length = 10  # 50
+    labels = torch.rand(num_classes, device=device)
+    inputs = torch.rand(N, 1, spectrogram_size, seq_length, device=device)
+    # Sequence length for each input
+    inputs_sizes = torch.rand(N, device=device).mul(seq_length * 0.1).add(seq_length * 0.8)
+    targets = torch.rand(N, target_length, device=device)
+    targets_sizes = torch.full((N,), target_length, dtype=torch.int, device=device)
```

Comment:
TODO(rzou): check how the random input generation works

Reply:
(resolved offline: I made a comment here https://github.com/pytorch/pytorch/pull/40586#discussion_r461797075 instead of filling in this TODO)